src/richie/apps/courses/models/course.py

```diff
@@ -473,6 +474,40 @@ def get_course_runs_dict(self):
 
         return dict(course_runs_dict)
 
+    def get_programs(self, language=None):
+        """
+        Return a query to get the programs related to this course ie for which a plugin for
+        this course is linked to the program page via any placeholder.
+        """
+        is_draft = self.extended_object.publisher_is_draft
+        course = self if is_draft else self.draft_extension
+        language = language or translation.get_language()
+
+        bfs = (
+            "extended_object__placeholders__cmsplugin__courses_coursepluginmodel__page"
+        )
+        filter_dict = {
+            "extended_object__publisher_is_draft": is_draft,
+            "extended_object__placeholders__cmsplugin__language": language,
+            bfs: course.extended_object,
+        }
+
+        program_model = apps.get_model(app_label="courses", model_name="program")
```

Comment:
Just for my understanding: Why not just import directly the Program model from `.program import Program` as it has been done for category, organization etc. ?

Reply:
I admit I copied it from somewhere else in the code. It is probably a relic of a generic code where the model name was passed as argument. I removed it to make it simpler.